### PR TITLE
[Alert activations] Add group by and aggregate by for alert activations

### PR DESCRIPTION
**Pull request by rokatyy**
Adds `AlertActivations` object to provide user ability to group and aggregate value. Inspired by the way these methods work in spark/pandas, so it be more intuitive for users who are familiar with them. 
 
Jira - https://iguazio.atlassian.net/browse/ML-8142

### CODE REVIEW

File: mlrun/common/schemas/alert.py [hunk 1]
```diff
@@ -216,3 +218,77 @@ class AlertActivation(pydantic.v1.BaseModel):
...
+        - If there's only one attribute, return the value directly (not a single-element tuple).
+        - If there are multiple attributes, return them as a tuple for grouping.
+
+        This ensures grouping behaves intuitively without redundant tuple representations.
+        """
+        if len(attributes) == 1:
+            # Avoid single-element tuple like (high,) when only one grouping attribute is used
+            return getattr(self, attributes[0])
+        # Otherwise, return a tuple of all specified attributes
+        return tuple(getattr(self, attr) for attr in attributes)
```
Comment (TomerShor) [inline mlrun/common/schemas/alert.py:234]: What about error handling? when the activation doesn't have one of the requested attribute?
Or maybe we are ok with `AttributeError`

File: mlrun/common/schemas/alert.py [hunk 2]
```diff
@@ -216,3 +218,77 @@ class AlertActivation(pydantic.v1.BaseModel):
     number_of_events: int
     notifications: list[notification_objects.NotificationState]
     reset_time: Optional[datetime] = None
+
+    def group_key(self, attributes: list[str]) -> tuple:
```
Comment (TomerShor) [inline mlrun/common/schemas/alert.py]: Is the return value a tuple even when `len(attributes) == 1`?

**Review by TomerShor — CHANGES_REQUESTED** [review]
That's a neat one :)

File: mlrun/common/schemas/alert.py (hunk 1 above)
Comment (rokatyy) [reply mlrun/common/schemas/alert.py:234]: @TomerShor I think it's ok to raise attribute error in this case

**Review by TomerShor — APPROVED** [review]

**Pull request merged**